platform/darwin/src/MGLStyleLayer.mm

```diff
@@ -65,6 +65,58 @@ - (float)minimumZoomLevel
     return self.rawLayer->getMinZoom();
 }
 
+- (void)setTransition:(MGLTransition)transition forKey:(NSString *)key
+{
+    SEL setPropertyTransitionSelector = [self mbx_selectorForKey:key type:@"set" hasParams:YES];
+    
+    if ([self respondsToSelector:setPropertyTransitionSelector]) {
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Warc-performSelector-leaks"
+        [self performSelector:setPropertyTransitionSelector withObject:[NSValue value:&transition withObjCType:@encode(MGLTransition)] ];
+#pragma clang diagnostic pop
+    }
+}
+
+- (MGLTransition)transitionForKey:(NSString *)key
+{
+    MGLTransition transition;
+    
+    SEL getPropertyTransitionSelector = [self mbx_selectorForKey:key type:@"get" hasParams:NO];
+    
+    if ([self respondsToSelector:getPropertyTransitionSelector]) {
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Warc-performSelector-leaks"
+        NSValue *transitionValue = [self performSelector:getPropertyTransitionSelector];
+        [transitionValue getValue:&transition];
+#pragma clang diagnostic pop
+    }
+    
+    return transition;
+}
+
+- (SEL)mbx_selectorForKey:(NSString *)key type:(NSString *)type hasParams:(BOOL)params
```

Comment:
Since all the valid setter method names form a closed set, we should generate a lookup table mapping attributes to the corresponding setter name, then look up the passed-in attribute in that table to decide what to call. That way, we can raise an exception when the developer passes in an untransitionable attribute or a bogus attribute. Currently, doing that leads to a nasty crash with no useful information.